pkg/shared/families/malware/yara/util/yara_util.go

```diff
@@ -0,0 +1,126 @@
+// Copyright © 2023 Cisco Systems, Inc. and its affiliates.
+// All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package util
+
+import (
+	"strings"
+
+	"github.com/sirupsen/logrus"
+
+	"github.com/openclarity/vmclarity/pkg/shared/families/malware/common"
+)
+
+const (
+	// A yara detected malware line should be in the following format: <yara rule name> [<coma,separated,meta data>] <path>.
+	scanError                   = "error scanning"
+	yaraRuleNamePosition        = 0
+	yaraRuleMetadataPosition    = 1
+	detectedMalwarePathPosition = 2
+	malwareNameString           = "name" // The key defined in the metadata of yara rule
+	malwareTypeString           = "type" // The key defined in the metadata of yara rule
+	yaraOutputFiledNum          = 3      // Number of yara output fields. <yara rule name> [<coma,separated,meta data>] <path>
+)
+
+func ParseYaraScanOutput(yaraOutput string, logger *logrus.Entry) []common.DetectedMalware {
+	logger.Debugf("Parsing yara output: %s", yaraOutput)
+	lines := strings.Split(yaraOutput, "\n")
+
+	malwareInfoList := make([]common.DetectedMalware, 0)
+	for i := range lines {
+		logger.Debugf("line %d: %s", i, lines[i])
+		if strings.Contains(lines[i], scanError) {
+			logger.Errorf("Yara scan failed: %s", lines[i])
+			continue
+		}
```

Comment:
@pbalogh-sa wdyt?

Reply:
We need to research it because there are no findings lines if nothing matches a rule, but permission errors can occur during opening files. So It is hard to find the right solution.

Reply:
ah I see, yeah thats tough... looks like we can force yara to output the files it scans using `--scan-list` so then we could compare it to the errors, but we'd need to fix the output streaming first to avoid blowing up the memory.

Reply:
Maybe worth opening another ticket for this to investigate